taichi/backends/cuda/codegen_cuda.cpp

```diff
@@ -398,18 +414,30 @@ class CodeGenLLVMCUDA : public CodeGenLLVM {
         // Issue an CUDA "__ldg" instruction so that data are cached in
         // the CUDA read-only data cache.
         auto dtype = stmt->ret_type;
-        auto llvm_dtype = llvm_type(dtype);
-        auto llvm_dtype_ptr = llvm::PointerType::get(llvm_type(dtype), 0);
-        llvm::Intrinsic::ID intrin;
-        if (is_real(dtype)) {
-          intrin = llvm::Intrinsic::nvvm_ldg_global_f;
+        if (auto ptr_type = stmt->ptr->ret_type->cast<PointerType>();
+            ptr_type->is_bit_pointer()) {
```

Comment:
Can `ptr_type` be `nullptr` here?

Reply:
Thanks for pointing this out! I will make it correct :-)